src/services/pcn-firewall/src/Chain.cpp

```diff
@@ -247,7 +235,7 @@ void Chain::updateChain() {
   std::map<int, std::vector<uint64_t>> protocols;
   std::vector<std::vector<uint64_t>> flags;
 
-  auto rules = getRealRuleList();
+  auto rules = rules_;
```

Comment:
Is this copy really needed?, can be rules_ be passed to `conntrack_from_rules_to_map`?

Reply:
You are right, I fixed that.